pkg/cli/start_jemalloc.go

```diff
@@ -19,7 +19,8 @@
 package cli
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
-// #cgo LDFLAGS: -ljemalloc
+// #cgo !darwin LDFLAGS: -ljemalloc
+// #cgo darwin LDFLAGS: -ljemalloc -u_je_zone_register
```

Comment:
>  before this change, jemalloc is not being used in our macOS builds, period.

Do we care other than the stats reporting? Btw, given that this is the second time this has broken, perhaps a small test is in order.